### PR TITLE
WindowCovering: Support of WNCV_2_3 + refactor of Mode/ConfigStatus

### PR DESCRIPTION
**Pull request by jmeg-sfy**
#### Problem
What is being fixed?  Examples:
* Update GoTos xml and callbacks to accommodate last specs update
* Rework Mode + ConfigStatus attributes using proper constant + BitFlags
* Add set function to Set Mode + ConfigStatus altogether
* Check if a device is operational/locked for motion by checking configstatus flags
* Add missing BitFlags test constraint template for YAML testing
* Add 1st implementation for YAML WNCV_2_3

#### Change overview
* Addition of WNCV_2_3 test using with function of Mode + ConfigStatus
* ZAP xml update to use BitFlags for mode + ConfigStatus
* Simplify initialization of ConfigStatus from the FeatureMap
* Update of All-cluster-app
* Update of EFR32 app

#### Testing
How was this tested? (at least one bullet point required)
* All previous Test cases are passing
* Tested on EFR32 demo
* Addition of new test WNCV_2_3

### CODE REVIEW

File: examples/window-app/common/src/WindowApp.cpp [hunk 1]
```diff
@@ -430,7 +438,13 @@ void WindowApp::Cover::Init(chip::EndpointId endpoint)
     EndProductTypeSet(endpoint, EndProductType::kInteriorBlind);
 
     // Attribute: Id 24 Mode
-    Mode mode = { .motorDirReversed = 0, .calibrationMode = 1, .maintenanceMode = 1, .ledDisplay = 1 };
+    chip::BitFlags<Mode> mode;
+    mode.Clear(Mode::kMotorDirectionReversed);
+    mode.Clear(Mode::kMaintenanceMode);
+    mode.Clear(Mode::kCalibrationMode);
+    mode.Set(Mode::kLedFeedback);
```
Comment (bzbarsky-apple) [inline examples/window-app/common/src/WindowApp.cpp:445]: This doesn't match the old code, which set the maintenance/calibration bits.  Is this a purposeful change?

File: src/app/clusters/window-covering-server/window-covering-server.cpp [hunk 2]
```diff
@@ -170,34 +170,45 @@ Type TypeGet(chip::EndpointId endpoint)
...
-    Attributes::ConfigStatus::Get(endpoint, &value);
-    status.operational             = (value & 0x01) ? 1 : 0;
-    status.online                  = (value & 0x02) ? 1 : 0;
-    status.liftIsReversed          = (value & 0x04) ? 1 : 0;
-    status.liftIsPA                = (value & 0x08) ? 1 : 0;
-    status.tiltIsPA                = (value & 0x10) ? 1 : 0;
-    status.liftIsEncoderControlled = (value & 0x20) ? 1 : 0;
-    status.tiltIsEncoderControlled = (value & 0x40) ? 1 : 0;
-    return status;
+    Attributes::ConfigStatus::Set(endpoint, configStatus);
```
Comment (bzbarsky-apple) [inline src/app/clusters/window-covering-server/window-covering-server.cpp:187]: OK to ignore failure here?

File: src/app/clusters/window-covering-server/window-covering-server.cpp [hunk 3]
```diff
@@ -249,23 +260,44 @@ EndProductType EndProductTypeGet(chip::EndpointId endpoint)
...
+
+    if (oldStatus != newStatus)
+        ConfigStatusSet(endpoint, newStatus);
+}
+
+chip::BitFlags<Mode> ModeGet(chip::EndpointId endpoint)
+{
+    chip::BitFlags<Mode> mode;
+
+    Attributes::Mode::Get(endpoint, &mode);
```
Comment (bzbarsky-apple) [inline src/app/clusters/window-covering-server/window-covering-server.cpp:300]: Again, ok to ignore error?

File: src/app/clusters/window-covering-server/window-covering-server.cpp [hunk 4]
```diff
@@ -619,6 +663,30 @@ void PostAttributeChange(chip::EndpointId endpoint, chip::AttributeId attributeI
         OperationalStatusSetWithGlobalUpdated(endpoint, opStatus);
 }
 
+EmberAfStatus GetMotionLockStatus(chip::EndpointId endpoint)
+{
+    BitFlags<Mode> mode                 = ModeGet(endpoint);
+    BitFlags<ConfigStatus> configStatus = ConfigStatusGet(endpoint);
+
+    // Does the device is locked ?
```
Comment (bzbarsky-apple) [inline src/app/clusters/window-covering-server/window-covering-server.cpp:671]:
```suggestion
    // Is the device locked?
```

File: src/app/clusters/window-covering-server/window-covering-server.cpp [hunk 5]
```diff
@@ -648,6 +718,15 @@ bool emberAfWindowCoveringClusterUpOrOpenCallback(app::CommandHandler * commandO
     EndpointId endpoint = commandPath.mEndpointId;
 
     emberAfWindowCoveringClusterPrint("UpOrOpen command received");
+
+    EmberAfStatus status = GetMotionLockStatus(endpoint);
+    if (EMBER_ZCL_STATUS_SUCCESS != status)
+    {
+        emberAfWindowCoveringClusterPrint("Err device locked");
+        emberAfSendImmediateDefaultResponse(status);
```
Comment (bzbarsky-apple) [inline src/app/clusters/window-covering-server/window-covering-server.cpp:726]: Followup: Please don't use `emberAfSendImmediateDefaultResponse`, which we are trying to get rid of.  Send responses via the command handler.

**Review by bzbarsky-apple — APPROVED** [review]

File: examples/window-app/common/src/WindowApp.cpp (hunk 1 above)
Comment (jmeg-sfy) [reply examples/window-app/common/src/WindowApp.cpp:445]: Old code was wrong and should not set those at init.

File: src/app/clusters/window-covering-server/window-covering-server.cpp (hunk 5 above)
Comment (jmeg-sfy) [reply src/app/clusters/window-covering-server/window-covering-server.cpp:726]: Didn't but good to know thanks

File: examples/window-app/common/src/WindowApp.cpp (hunk 1 above)
Comment (jmeg-sfy) [reply examples/window-app/common/src/WindowApp.cpp:445]: This really depend of the init state of a devices and it is manufacturer specific
For the vanilla demo everything should be ready to be used

File: src/app/clusters/window-covering-server/window-covering-server.cpp (hunk 2 above)
Comment (jmeg-sfy) [reply src/app/clusters/window-covering-server/window-covering-server.cpp:187]: yes

File: src/app/clusters/window-covering-server/window-covering-server.cpp (hunk 3 above)
Comment (jmeg-sfy) [reply src/app/clusters/window-covering-server/window-covering-server.cpp:300]: yes

Comment (jmeg-sfy) [reply src/app/clusters/window-covering-server/window-covering-server.cpp:300]: the only risk is to have wrong ep problem

**Review by andy31415 — APPROVED** [review]

**Pull request merged**